lib/sensors/temperature.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "i2c_sensors.hpp"
+
+#include <cstdint>
+#include <cstdio>
+#include <optional>
+
+#include <core/logger.hpp>
+#include <io/i2c.hpp>
+
+namespace hyped::sensors {
+
+static constexpr std::uint8_t kCtrl   = 0x04;
+static constexpr std::uint8_t kDataTH = 0x07;
+static constexpr std::uint8_t kDataTL = 0x06;
+static constexpr std::uint8_t kStatus = 0x05;
+// Sets the sensor to continuous mode, sets IF_ADD_INC, and sets sampling rate to 200Hz
+static constexpr std::uint8_t kConfigurationSetting = 0x3c;
+
+class Temperature : public II2cMuxSensor<std::int16_t> {
+ public:
+  Temperature(const std::uint8_t device_address,
+              const std::uint8_t channel,
+              io::I2c &i2c,
+              hyped::core::ILogger &log);
```

Comment:
I don't like the ordering here; can we put the logger, then I2c and then everything else? basically the rule should be more common things to the front and more specialised things to the back

Reply:
i think this is the case with other places in the codebase too, will fix some in my mux PR

Reply:
fixed by #59

Reply:
@HTTYDKing quick fix on this + run clang format before commiting